src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,334 @@
+        /// try to process file contents.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="literalPath"></param>
+        /// <returns></returns>
+        private List<string> ResolvePaths(string[] path, bool literalPath)
+        {
+            List<string> pathsToProcess = new List<string>();
+            ProviderInfo provider = null;
+            PSDriveInfo drive = null;
+
+            foreach (string currentPath in path)
+            {
+                List<string> newPaths = new List<string>();
+
+                if (literalPath)
+                {
+                    newPaths.Add(Context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(currentPath, out provider, out drive));
+                }
+                else
+                {
+                    try
+                    {
+                        newPaths.AddRange(Context.SessionState.Path.GetResolvedProviderPathFromPSPath(currentPath, out provider));
+                    }
+                    catch (ItemNotFoundException e)
+                    {
+                        if (!WildcardPattern.ContainsWildcardCharacters(currentPath))
+                        {
+                            ErrorRecord errorRecord = new ErrorRecord(e, "FileNotFound", ErrorCategory.ObjectNotFound, path);
+                            WriteError(errorRecord);
+                            continue;
+                        }
+                    }
+                }
+
+                if (!provider.Name.Equals("FileSystem", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Write a non-terminating error message indicating that path specified is not supported.
+                    string errorMessage = StringUtil.Format(UtilityCommonStrings.FormatHexOnlySupportsFileSystemPaths, currentPath);
+                    ErrorRecord errorRecord = new ErrorRecord(new ArgumentException(errorMessage),
+                                                                "FormatHexOnlySupportsFileSystemPaths",
+                                                                ErrorCategory.InvalidArgument,
+                                                                currentPath);
+                    WriteError(errorRecord);
+                    continue;
+                }
+
+                pathsToProcess.AddRange(newPaths);
+            }
+
+            return pathsToProcess;
+        }
+
+        /// <summary>
+        /// Pass each valid path on to process its contents.
+        /// </summary>
+        /// <param name="pathsToProcess"></param>
+        private void ProcessPath(List<string> pathsToProcess)
+        {
+            // If there are no paths to proccess, just return.
+            if (pathsToProcess.Count == 0)
+            {
+                return;
+            }
+
+            foreach (string path in pathsToProcess)
+            {
+                ProcessFileContent(path);
```

Comment:
There is a header for each ByteCollection object, e.g., 

New-Item -Value "foo" -Path foo1.txt -ItemType File
New-Item -Value "foo2" -Path foo2.txt -ItemType File

PS C:\exp> dir foo* | fhx
           Path: C:\exp\foo1.txt
           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
00000000   66 6F 6F                                         foo

           Path: C:\exp\foo2.txt
           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
00000000   66 6F 6F 32                                      foo2

PS C:\exp> fhx -Path .\foo*
           Path: C:\exp\foo1.txt
           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
00000000   66 6F 6F                                         foo

           Path: C:\exp\foo2.txt
           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
00000000   66 6F 6F 32                                      foo2